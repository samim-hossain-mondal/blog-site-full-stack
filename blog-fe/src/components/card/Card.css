img {
    width: 100%;
    max-width: 100%;
    height: 250px;
    object-fit: cover;
}

.card {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    row-gap: 35px;
    column-gap: 100px;
    padding: 50px;
}

.container {
    display: flex;
    flex-direction: column;
    padding: 10px;
}

.card-blog {
    display: flex;
    flex-direction: column;
    width: 320px;
    border: solid rgb(200, 200, 200) 1px;
}

.date-time {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    font-size: 12px;
    color: rgb(125, 125, 125);
}

.content-head {
    font-size: 25px;
    padding-top: 10px;
    padding-bottom: 10px;
    inline-size: inherit;
    overflow-wrap: break-word;
}

.content-body {
    font-size: 15px;
    inline-size: inherit;
    overflow-wrap: break-word;
    color: rgb(100, 100, 100);
}

.icons {
    padding: 5px;
    width: 23px;
    height: 23px;
    display: flex;
    align-self: flex-end;
}

.card-foot {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}