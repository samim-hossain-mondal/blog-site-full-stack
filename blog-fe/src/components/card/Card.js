/* eslint-disable no-undef */
import React from 'react';
import Clap from '../clap/Clap.js';
import Like from '../like/Like.js';
import cardData from '../../assets/mockData/index.js';
import './Card.css';

export default function Card() {
  const card = cardData.map((card, index) => {
    return (<div key={index} className='card-blog'>
      <img src={require(`../../assets/images/${card.image}`)} />
      <div className='container'>
        <div className='date-time'>
          <h> {card.date} </h>
          <h> {card.readingTime} </h>
        </div>
        <div className='content-head'>
          <h> {card.title} </h>
        </div>
        <div className='content-body'>
          <h> {card.description} </h>
        </div>
        <div>
          <hr />
          <div className='card-foot'>
            <Clap claps={card.claps} />
            <Like />
          </div>
        </div>
      </div>
    </div>);
  });
  return (
    <div className='card'>
      {card}
    </div>
  );
}