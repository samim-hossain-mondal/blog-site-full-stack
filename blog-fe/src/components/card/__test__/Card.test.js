/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import Card from '../Card';

describe('Card', () => {
  it('should render', () => {
    const { asFragment } = render(<Card />);
    expect(asFragment()).toMatchSnapshot();
  });
});