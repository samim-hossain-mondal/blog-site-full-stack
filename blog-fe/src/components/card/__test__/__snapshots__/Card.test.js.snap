// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-blog"
    >
      <img
        src="abstract.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             2nd Januray, 2018 
          </h>
          <h>
             2 mins 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             The future of abstract art and the culture ... 
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 10 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-blog"
    >
      <img
        src="outdoor.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             31st Januray, 2018 
          </h>
          <h>
             1 min 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             How painting outdoors can transform your  
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 1 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-blog"
    >
      <img
        src="young-painters.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             20th March, 2018 
          </h>
          <h>
             2 mins 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             10 young painters you need to know 
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 15 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-blog"
    >
      <img
        src="perfect-art.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             22nd June, 2018 
          </h>
          <h>
             5 mins 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             Tips for creating the perfect art 
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 11 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-blog"
    >
      <img
        src="inspired.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             4th Feburary, 2019 
          </h>
          <h>
             9 mins 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             Stay inspired: top artists blog ... 
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 45 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card-blog"
    >
      <img
        src="reason.png"
      />
      <div
        class="container"
      >
        <div
          class="date-time"
        >
          <h>
             10th Januray, 2019 
          </h>
          <h>
             2 mins 
          </h>
        </div>
        <div
          class="content-head"
        >
          <h>
             5 reasons why everyone should try an... 
          </h>
        </div>
        <div
          class="content-body"
        >
          <h>
             Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your... 
          </h>
        </div>
        <div>
          <hr />
          <div
            class="card-foot"
          >
            <div
              class="clap"
            >
              <img
                class="icons"
                data-testid="clap-btn"
                src="clapping.svg"
                style="opacity: 0.35;"
              />
              <h>
                 11 
              </h>
            </div>
            <div
              class="like"
            >
              <img
                class="icons"
                data-testid="like-btn"
                src="heart-black.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
